### PR TITLE
Make exemplar default language normative

### PR DESCRIPTION
**Pull request by alanwest**
I'm making my best guess of what was intended. If the language should use `SHOULD` or `MAY` please comment.

### CODE REVIEW

File: specification/metrics/sdk.md [hunk 1]
```diff
@@ -1105,13 +1105,13 @@ The SDK MUST include two types of built-in exemplar reservoirs:
 
 By default:
 
-- Explicit bucket histogram aggregation with more than 1 bucket will
+- Explicit bucket histogram aggregation with more than 1 bucket MUST
```
Comment (reyang) [inline specification/metrics/sdk.md:1108]: How would this change be interpreted? The next paragraph said that reservoirs MAY change and that's not a breaking change.

**Comment by reyang** [comment]
> If the language should use `SHOULD` or `MAY` please comment.

+1 on SHOULD.

File: specification/metrics/sdk.md (hunk 1 above)
Comment (alanwest) [reply specification/metrics/sdk.md:1108]: Good question, I'm not sure.

Based on @jack-berg's comment https://github.com/open-telemetry/opentelemetry-specification/pull/3943#issuecomment-2066957077, my interpretation is that the spec could make the following change in the future without considering it breaking:

```diff
- Explicit bucket histogram aggregation with more than 1 bucket MUST use `AlignedHistogramBucketExemplarReservoir`.
+ Explicit bucket histogram aggregation with more than 1 bucket MUST use `NewAndImprovedExemplarReservoir`.
```

Comment (reyang) [reply specification/metrics/sdk.md:1108]: No strong opinion here (either approach sounds okay to me).

If you read line 1110, "Base2 Exponential Histogram Aggregation SHOULD use a ..." is using "SHOULD" as well, I don't see a compelling reason to use MUST.

Comment (alanwest) [reply specification/metrics/sdk.md]: Changed to SHOULD

**Review by cijothomas — APPROVED** [review]
SHOULD is reasonable here.

**Review by MrAlias — APPROVED** [review]

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**